CRM/Contribute/Page/Tab.php

```diff
@@ -73,15 +73,18 @@ public static function recurLinks($recurID = FALSE, $context = 'contribution') {
       if ($paymentProcessorObj) {
         if ($paymentProcessorObj->supports('cancelRecurring')) {
           unset($links[CRM_Core_Action::DISABLE]['extra'], $links[CRM_Core_Action::DISABLE]['ref']);
-          $links[CRM_Core_Action::DISABLE]['url'] = "civicrm/contribute/unsubscribe";
+          $links[CRM_Core_Action::DISABLE]['url'] = $paymentProcessorObj->subscriptionURL(NULL, NULL, 'cancel');
```

Comment:
@mattwire @webmaster-cses-org-uk this link would be available to backoffice users and to people viewing their own dashboard.

I think the desired link & behaviour might be different  - ie the front end user should be taken off the paypal whereas the backoffice user can only do an administrative cancel (which shouldn't be available to the front end user).

I have a feeling the code might be simpler if we didn't have one 'recurLinks' function try to deal with both scenarios

Reply:
I agree splitting it is probably better. It would be cleaner and easier to maintain: relying on the free-text $context argument leaves a bit too much to chance I think.

One other thing to consider: among the changes I've proposed, supports("cancelRecurring") will return FALSE for PayPal_Standard. This is because the method is currently defined (in the comments) as whether the processor supports cancellation "in code", which of course PayPal_Standard doesn't. This would mean that no link is present for back-office or front-end users.

I think we may therefore also need to split the "cancelRecurring" operation into two: back-office and front-end, and similarly have two modes for subscriptionURL (back-end and front-end cancel).

Reply:
Yes - I think so - maybe we should supportUserCancelRecurring (or supportDonorCancelRecurring)

Reply:
I'd second that. Either of those names would be fine. Then for subscriptionURL perhaps we could have a corresponding 'cancelUser' (or 'cancelDonor')?